src/source/Ice/SocketConnection.c

```diff
@@ -35,6 +35,7 @@ STATUS createSocketConnection(KVS_IP_FAMILY_TYPE familyType, KVS_SOCKET_PROTOCOL
     }
     ATOMIC_STORE_BOOL(&pSocketConnection->connectionClosed, FALSE);
     ATOMIC_STORE_BOOL(&pSocketConnection->receiveData, FALSE);
+    ATOMIC_STORE(&pSocketConnection->inUse, FALSE);
```

Comment:
should use ATOMIC_STORE_BOOL here as well

Reply:
D'oh. Will change.

BTW. It seems that we do have TSAN exception for this particular failure case but for some reason it doesn't get picked up. Need to re-run